woocommerce/woocommerce
# [e2e tests] Add downloadable product test

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes https://github.com/woocommerce/woocommerce-quality/issues/612 .

Cover the add downloadable product flow. 

Changes:
- renamed the `create-simple-product.spec.js` spec file to `product-create-simple.spec.js` to be consistent with the pattern `product-<action>` used by most spec files (also updated the `docs/quality-and-best-practices/core-critical-flows.md` doc as an effect).
- I used a for loop to merge the resulting 3 tests into a single parametrised test function to reduce code duplication. My first intention was to have 3 tests and extract the common parts into functions, but I think in the end this version is cleaner and more readable.
- With merging all tests into one function I added some extra checks for the virtual and non-virtual products.
- I also enabled the `fullPage` option for screenshots.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

Run the spec locally with `pnpm end:restart && pnpm test:e2e-pw ./tests/e2e-pw/tests/merchant/product-create-simple.spec.js`

⚠️ The `MarkdownLint on PR / lint` check is failing because I updated a doc file and the check asks for an update of the docs-manifest, but this is not required for updates of content only.



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
